### ISSUE TITLE
Composer install --no-dev failing with sf4 + flex

### ISSUE THREAD

**Issue by djamadeus** [EasyCorp/easy-deploy-bundle#35]
I fixed this here, 
https://github.com/EasyCorp/easy-deploy-bundle/commit/45da26d38d1281ea9ff201bfaa03e854fbb5f64a

messed up the fork though, so can't create clean pullrequest (or at least I dunno  know how)

**Comment by djamadeus** [comment]
this one hast it cleaner:
https://github.com/Wunderdata/easy-deploy-bundle-wunderfork/commit/35a3a31325e84d9e961723ab2779ab6df468add0

**Comment by zorn-v** [comment]
It is not bundle problem. You can move symfony/dotenv from dev deps, or just set env vars properly.
http://symfony.com/doc/current/best_practices/configuration.html

**Comment by djamadeus** [comment]
yep but you still have pass app_env manually to the "composer install"-call, otherwise the call will fail with
"symfony-cmd handling the auto-scripts event returned with error code 127"

**Comment by zorn-v** [comment]
You can set it (and other vars like APP_DEBUG) in `~/.profile` or `~/.bashrc` for user which doing deploy for example `export APP_ENV=prod`
Or just
```bash
set -a
. /path/to/project/.env
set +a
```
https://unix.stackexchange.com/questions/79064/how-to-export-variables-from-a-file

**Comment by djamadeus** [comment]
Yep true, thanks for the hint. I'm switching between prod and debug on
staging server though, so adding this directly to import command, using
value from symfony env, still makes sense for me, (and keeps config in one
place centralized, and makes composer use the env chosen in deploy-config,
regardless from server config) but I agree, this is no bug/issue of the
module then.




Am 06.12.2017 00:59 schrieb "zorn-v" <notifications@github.com>:

You can set it (and other vars like APP_DEBUG) in ~/.profile or ~/.bashrc
for user which doing deploy for example export APP_ENV=prod
Or just

set -a. /path/to/project/.envset +a

https://unix.stackexchange.com/questions/79064/how-to-
export-variables-from-a-file

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/EasyCorp/easy-deploy-bundle/issues/35#issuecomment-349482489>,
or mute the thread

**Comment by bzoks** [comment]
AbstractDeployer actually sets variable SYMFONY_ENV to provided ->symfonyEnvironment(), which is later used in commands:
`[xxx@server.com] Executing command: (export SYMFONY_ENV=staging; cd /usr/local/www/...]`
but this environment variable has no meaning in Symfony4 - code should set APP_ENV instead when run on version 4.

**Comment by cve** [comment]
I also have problems with Symfony 4 and .env file

**Comment by MSDATDE** [comment]
@cve Have you found the solution to this problem? I have the same problem with Symfony 4.

**Comment by cve** [comment]
@MSDATDE not yet

**Comment by cve** [comment]
Does anyone has an example how to using this bundle with Symfony 4?

**Comment by deantomasevic** [comment]
Hi,

i had the same error:

```
  Generating autoload files
  Executing script cache:clear [KO]
   [KO]
  Script cache:clear returned with error code 255
  !!  PHP Fatal error:  Uncaught Symfony\Component\Dotenv\Exception\PathException: Unable to read the "/var/www/dean.io/releases/20190126225728/.env" environment file. in /var/www/dean.io/releases/20190126225728/vendor
  /symfony/dotenv/Dotenv.php:466
  !!  Stack trace:
```

The main problem is: https://symfony.com/doc/current/configuration/dot-env-changes.html

> B) The .env file is now commited to your repository. It was previously ignored via the .gitignore file (the updated recipe does not ignore this file). Because this file is committed, it should contain non-sensitive, default values. Basically, the .env.dist file was moved to .env.

FIX: https://github.com/EasyCorp/easy-deploy-bundle/pull/86

**Comment by PtrTn** [comment]
I ran into the same issue. Since #86 has not been merged yet I solved it by adding a command to the `beforePreparing()` hook as such:
```
public function beforePreparing()
{
   $this->runRemote(sprintf('cp {{ deploy_dir }}/repo/.env {{ project_dir }} 2>/dev/null'));
}
```

**Comment by COil** [comment]
@PtrTn  I have made the same but still having an error afer composer install:

```
[KO]
  Script cache:clear returned with error code 1
  !!
  !!  In EnvVarProcessor.php line 162:
  !!
  !!    Environment variable not found: "APP_SERVER".
```
